crates/block/header/src/lib.rs

```diff
@@ -205,21 +205,21 @@ impl<HardforkT: Into<EvmSpecId>> BlockEnvForHardfork<HardforkT> for BlockHeader
 /// Wrapper type combining a header with its associated hardfork.
 ///
 /// Both are needed to implement the [`BlockEnvTrait`] trait.
-pub struct HeaderAndEvmSpec<'header, BlockHeaderT: BlockEnvForHardfork<HardforkT>, HardforkT> {
+pub struct HeaderAndEvmSpec<'spec, BlockHeaderT: BlockEnvForHardfork<HardforkT>, HardforkT> {
     pub hardfork: HardforkT,
-    pub header: &'header BlockHeaderT,
-    pub scheduled_blob_params: Option<ScheduledBlobParams>,
+    pub header: &'spec BlockHeaderT,
+    pub scheduled_blob_params: Option<&'spec ScheduledBlobParams>,
```

Comment:
Suggestion: for consistency

```suggestion
    pub header: &'env BlockHeaderT,
    pub scheduled_blob_params: Option<&'env ScheduledBlobParams>,
```